csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -53,5 +54,24 @@ public sealed class ArrayData
             Buffers = buffers;
             Children = children;
         }
+
+        public void Dispose()
+        {
+            if (Buffers != null)
+            {
+                foreach (ArrowBuffer buffer in Buffers)
+                {
+                    buffer.Dispose();
+                }
+            }
+
+            if (Children != null)
```

Comment:
It's just used in these 2 places, is it really worth an extra method? Also, in one case it can have nulls and the other it can't (because `ArrowBuffer` is a `struct`).

Reply:
In 3, the nulls can be handled too with `?` (the generic implementation for struct should remove the null check) 

https://github.com/apache/arrow/blob/37a9b187276a5a68760becb816faeb0d44b00ac2/csharp/src/Apache.Arrow/RecordBatch.cs#L56-L58